projects/gloo/pkg/plugins/kubernetes/eds.go

```diff
@@ -203,50 +207,54 @@ func (c *edsWatcher) watch(writeNamespace string, opts clients.WatchOpts) (<-cha
 	return endpointsChan, errs, nil
 }
 
+type Epkey struct {
+	Address     string
+	Port        uint32
+	Name        string
+	Namespace   string
+	UpstreamRef *core.ResourceRef
+}
+
+func findPortForService(services []*kubev1.Service, spec *kubeplugin.UpstreamSpec) (*kubev1.ServicePort, bool) {
+	for _, svc := range services {
+		if svc.Namespace != spec.GetServiceNamespace() || svc.Name != spec.GetServiceName() {
+			continue
+		}
+		if len(svc.Spec.Ports) == 1 {
+			if spec.GetServicePort() == uint32(svc.Spec.Ports[0].Port) {
+				return &svc.Spec.Ports[0], true
+			}
+		}
+		for _, port := range svc.Spec.Ports {
+			if spec.GetServicePort() == uint32(port.Port) {
+				return &port, false
+			}
+		}
```

Comment:
Since we are already reframing this code we could consider calling out the fact that we return whether this is a single port service.
Either as a method comment and/or doing something like the following

```suggestion
		for _, port := range svc.Spec.Ports {
			if spec.GetServicePort() == uint32(port.Port) {
				return &port,  (len(svc.Spec.Ports) == 1)
			}
		}
```